flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -467,6 +487,128 @@ public void registerAtResourceManager(final String address) {
 		//TODO:: register at the RM
 	}
 
+	@RpcMethod
+	public NextInputSplit requestNextInputSplit(final JobVertexID vertexID, final ExecutionAttemptID executionAttempt) {
```

Comment:
Let's let this method throw an exception. Then the calling TaskManager can see the difference between `null` (= no further split) and `Exception` (= something went wrong / is inconsistent).